grants/views.py

```diff
@@ -0,0 +1,112 @@
+from django.shortcuts import render
+from django.contrib import messages
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.shortcuts import redirect
+from django.urls import reverse_lazy
+from django.views.generic import CreateView, UpdateView, ListView, DetailView
+from django.utils import timezone
+
+from .models import GrantApplication
+from .forms import GrantApplicationForm, GrantReviewForm
+
+# Create your views here.
+
+class GrantApplicationCreateView(LoginRequiredMixin, CreateView):
+    model = GrantApplication
+    form_class = GrantApplicationForm
+    template_name = 'grants/application_form.html'
+    success_url = reverse_lazy('grants:application_detail')
+
+    def form_valid(self, form):
+        form.instance.user = self.request.user
+        messages.success(self.request, 'Your grant application has been submitted successfully!')
+        return super().form_valid(form)
+
+    def get(self, request, *args, **kwargs):
+        # Check if user already has an application
+        if hasattr(request.user, 'grant_application'):
+            messages.warning(request, 'You have already submitted a grant application.')
+            return redirect('grants:application_detail')
+        return super().get(request, *args, **kwargs)
+
+class GrantApplicationUpdateView(LoginRequiredMixin, UserPassesTestMixin, UpdateView):
```

Comment:
The accept/decline logic will only come into play later on. Wont review that yet.

It's better to keep PRs small - the issue is that if there is a bug in any of the code that isn't necessary yet, then that holds up the merge.

A small PR can be reviewed much faster and is less likely to need extra cycles of reviews.